csrc/root_domain_map.cpp

```diff
@@ -121,6 +121,34 @@ std::unordered_map<IterDomain*, IterDomain*> PairwiseRootDomainMap::map(
   const auto producer_root =
       TensorDomain::noReductions(producer->maybeRFactor());
   const auto& consumer_root = consumer->root();
+
+  // Map MatmulOp extents in reverse
+  if (consumer_tv_->definition()->isA<MatmulOp>()) {
```

Comment:
Q: Do we need refactor this special handling in a separate override method, for instance?

Reply:
Possibly, but I don't think that's necessary since `PairwiseRootDomainMap` is not subclassed. Its parent class is `RootDomainMap` which does not need this functionality. Hypothetically could publicly inherit `OptOutConstDispatch` possibly and call `dispatch(consumer_tv_->definition())` here. However, we wouldn't know which input and output are this particular producer and consumer. So we'd wind up reinventing a bunch of stuff I think. Since it's not needed anywhere but here, I think it's safe to keep these `if`s here and just make them as tidy as possible.